cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -122,6 +122,23 @@ def get_slurm_specific_dna_json_for_head_node(config: SlurmClusterConfig, schedu
     }
 
 
+def get_slurm_database_dna_json_for_head_node(config: BaseClusterConfig) -> dict:
```

Comment:
why we need this to explicitly go into the dna.json and can't only rely on the config?

Reply:
At first I thought the dna.json was the right way to export parameters to make them available to Chef (because it's ubiquitous in the code). After your comment I dug a bit more in the cookbook and found [this](https://github.com/aws/aws-parallelcluster-cookbook/blob/da0302f6a491987dc35b078a3a9e3f0a89d61f4a/cookbooks/aws-parallelcluster-config/recipes/config.rb#L18-L22):

```chef
include_recipe 'aws-parallelcluster-config::base'

fetch_config 'Fetch and load cluster configs' unless node['cluster']['scheduler'] == 'awsbatch'

include_recipe 'aws-parallelcluster-slurm::config' if node['cluster']['scheduler'] == 'slurm'
```

which I guess is why all the operations performed in `base.rb` currently rely on parameters defined in either `defaults.rb` or exported to the `dna.json` (e.g. directory service).

Since the Slurm operations happen after the config fetching operation I can now remove what I had exported into the `dna.json` and use the imported structure. Is there a reason we do this only after the `base.rb` recipe has been called?

Reply:
Actually, we do need at least a trigger (`['slurm']['database']['enabled']) for the Cloudwatch agent: https://github.com/aws/aws-parallelcluster-cookbook/pull/1532/ .

@demartinofra @davprat maybe we should just remove the `feature_conditions` for the export of `/var/log/slurmdbd.log`?

Reply:
Done.

Reply:
Fdm says it's OK

Reply:
Done.